File: backend/app/models/assistant.py
from pydantic import BaseModel
from typing import List, Literal, Optional


class ChatMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str


class ChatRequest(BaseModel):
    message: str
    history: List[ChatMessage] = []


class ReportFile(BaseModel):
    filename: str
    content_base64: str


class ChatResponse(BaseModel):
    reply: str
    report: Optional[ReportFile] = None

File: frontend/src/api/assistant.js
const API_URL = import.meta.env.VITE_API_URL || 'http://localhost:8000';

// `history` is [{ role: 'user' | 'assistant', content: string }, ...] —
// only the earlier turns, not the message being sent now.
// `token` is optional: pass it when the visitor is logged in so the backend
// can tell whether they're an admin (needed for the sales-report tool).
export async function sendChatMessage(message, history = [], token) {
  const res = await fetch(`${API_URL}/assistant/chat`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      ...(token ? { Authorization: `Bearer ${token}` } : {}),
    },
    body: JSON.stringify({ message, history }),
  });
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error(err.detail || 'The assistant is unavailable right now.');
  }
  return res.json();
}